Improve Quarkus native support

Improve Quarkus native support with extensions based on other project experiences.

- fix etcd error messages, e.g.: .`..Caused by: java.lang.IllegalStateException: Generated message class "io.etcd.jetcd.api.RangeResponse" missing method "getHeader...`
- fix jsonb FieldOnlyVisibilityStrategy error messages, e.g.: `...Caused by: java.lang.ClassNotFoundException: hu.icellmobilsoft.coffee.tool.jsonb.FieldOnlyVisibilityStrategy...`

--- a/coffee-bom/coffee-bom-project/pom.xml
+++ b/coffee-bom/coffee-bom-project/pom.xml
@@ -313,7 +313,19 @@
 
             <dependency>
                 <groupId>hu.icellmobilsoft.coffee</groupId>
+                <artifactId>coffee-module-etcd-extension</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+
+            <dependency>
+                <groupId>hu.icellmobilsoft.coffee</groupId>
                 <artifactId>coffee-module-etcd-extension-deployment</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+
+            <dependency>
+                <groupId>hu.icellmobilsoft.coffee</groupId>
+                <artifactId>coffee-tool-extension</artifactId>
                 <version>${project.version}</version>
             </dependency>
 

--- a/coffee-bom/coffee-bom-project/pom.xml
+++ b/coffee-bom/coffee-bom-project/pom.xml
@@ -310,6 +310,18 @@
                 <artifactId>coffee-deltaspike-data-extension-deployment</artifactId>
                 <version>${project.version}</version>
             </dependency>
+
+            <dependency>
+                <groupId>hu.icellmobilsoft.coffee</groupId>
+                <artifactId>coffee-module-etcd-extension-deployment</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+
+            <dependency>
+                <groupId>hu.icellmobilsoft.coffee</groupId>
+                <artifactId>coffee-tool-extension-deployment</artifactId>
+                <version>${project.version}</version>
+            </dependency>
         </dependencies>
     </dependencyManagement>
 </project>

--- a/coffee-quarkus-extensions/coffee-deltaspike-data-extension-parent/pom.xml
+++ b/coffee-quarkus-extensions/coffee-deltaspike-data-extension-parent/pom.xml
@@ -19,39 +19,4 @@
         <module>coffee-deltaspike-data-extension</module>
     </modules>
 
-    <build>
-        <pluginManagement>
-            <plugins>
-                <plugin>
-                    <groupId>io.quarkus</groupId>
-                    <artifactId>quarkus-maven-plugin</artifactId>
-                    <version>${quarkus.version}</version>
-                </plugin>
-                <plugin>
-                    <artifactId>maven-surefire-plugin</artifactId>
-                    <configuration>
-                        <systemPropertyVariables>
-                            <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
-                        </systemPropertyVariables>
-                    </configuration>
-                </plugin>
-                <plugin>
-                    <artifactId>maven-failsafe-plugin</artifactId>
-                    <configuration>
-                        <systemPropertyVariables>
-                            <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
-                        </systemPropertyVariables>
-                    </configuration>
-                </plugin>
-                <plugin>
-                    <artifactId>maven-compiler-plugin</artifactId>
-                    <configuration>
-                        <compilerArgs>
-                            <arg>-parameters</arg>
-                        </compilerArgs>
-                    </configuration>
-                </plugin>
-            </plugins>
-        </pluginManagement>
-    </build>
 </project>

--- a/coffee-quarkus-extensions/coffee-module-etcd-extension-parent/coffee-module-etcd-extension-deployment/pom.xml
+++ b/coffee-quarkus-extensions/coffee-module-etcd-extension-parent/coffee-module-etcd-extension-deployment/pom.xml
@@ -1,0 +1,44 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>hu.icellmobilsoft.coffee</groupId>
+        <artifactId>coffee-module-etcd-extension-parent</artifactId>
+        <version>2.11.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>coffee-module-etcd-extension-deployment</artifactId>
+    <name>Coff:ee module etcd Extension - Deployment</name>
+
+    <dependencies>
+
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-arc-deployment</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>hu.icellmobilsoft.coffee</groupId>
+            <artifactId>coffee-module-etcd</artifactId>
+        </dependency>
+
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <annotationProcessorPaths>
+                        <path>
+                            <groupId>io.quarkus</groupId>
+                            <artifactId>quarkus-extension-processor</artifactId>
+                            <version>${quarkus.version}</version>
+                        </path>
+                    </annotationProcessorPaths>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+</project>

--- a/coffee-quarkus-extensions/coffee-module-etcd-extension-parent/coffee-module-etcd-extension-deployment/src/main/java/hu/icellmobilsoft/coffee/quarkus/extension/module/etcd/deployment/EtcdReflectionProcessor.java
+++ b/coffee-quarkus-extensions/coffee-module-etcd-extension-parent/coffee-module-etcd-extension-deployment/src/main/java/hu/icellmobilsoft/coffee/quarkus/extension/module/etcd/deployment/EtcdReflectionProcessor.java
@@ -41,16 +41,21 @@ public class EtcdReflectionProcessor {
     private static final Logger log = Logger.getLogger(EtcdReflectionProcessor.class);
     private static final String FEATURE = "coffee-module-etcd-extension";
 
-    @BuildStep
-    FeatureBuildItem feature() {
-        return new FeatureBuildItem(FEATURE);
-    }
-
     /**
      * No-arg constructor
      */
     public EtcdReflectionProcessor() {
         // No-arg constructor for java 21
+    }
+
+    /**
+     * Registers feature name
+     *
+     * @return Feature build item
+     */
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
     }
 
     /**

--- a/coffee-quarkus-extensions/coffee-module-etcd-extension-parent/coffee-module-etcd-extension-deployment/src/main/java/hu/icellmobilsoft/coffee/quarkus/extension/module/etcd/deployment/EtcdReflectionProcessor.java
+++ b/coffee-quarkus-extensions/coffee-module-etcd-extension-parent/coffee-module-etcd-extension-deployment/src/main/java/hu/icellmobilsoft/coffee/quarkus/extension/module/etcd/deployment/EtcdReflectionProcessor.java
@@ -27,6 +27,7 @@ import io.etcd.jetcd.api.KeyValue;
 import io.etcd.jetcd.api.RangeResponse;
 import io.etcd.jetcd.api.ResponseHeader;
 import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
 import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
 
 /**
@@ -38,12 +39,18 @@ import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
 public class EtcdReflectionProcessor {
 
     private static final Logger log = Logger.getLogger(EtcdReflectionProcessor.class);
+    private static final String FEATURE = "coffee-module-etcd-extension";
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
+    }
 
     /**
-     * Default constructor
+     * No-arg constructor
      */
     public EtcdReflectionProcessor() {
-        // Default constructor for java 21
+        // No-arg constructor for java 21
     }
 
     /**

--- a/coffee-quarkus-extensions/coffee-module-etcd-extension-parent/coffee-module-etcd-extension-deployment/src/main/java/hu/icellmobilsoft/coffee/quarkus/extension/module/etcd/deployment/EtcdReflectionProcessor.java
+++ b/coffee-quarkus-extensions/coffee-module-etcd-extension-parent/coffee-module-etcd-extension-deployment/src/main/java/hu/icellmobilsoft/coffee/quarkus/extension/module/etcd/deployment/EtcdReflectionProcessor.java
@@ -1,0 +1,61 @@
+/*-
+ * #%L
+ * Coffee
+ * %%
+ * Copyright (C) 2020 - 2025 i-Cell Mobilsoft Zrt.
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package hu.icellmobilsoft.coffee.quarkus.extension.module.etcd.deployment;
+
+import java.util.List;
+
+import org.jboss.logging.Logger;
+
+import io.etcd.jetcd.api.KeyValue;
+import io.etcd.jetcd.api.RangeResponse;
+import io.etcd.jetcd.api.ResponseHeader;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+
+/**
+ * Registers necessary etcd related classes for reflection to ensure compatibility with Quarkus native
+ *
+ * @author janos.hamrak
+ * @since 2.11.0
+ */
+public class EtcdReflectionProcessor {
+
+    private static final Logger log = Logger.getLogger(EtcdReflectionProcessor.class);
+
+    /**
+     * Default constructor
+     */
+    public EtcdReflectionProcessor() {
+        // Default constructor for java 21
+    }
+
+    /**
+     * Register jetcd classes for reflection.
+     *
+     * @return registered classes
+     */
+    @BuildStep
+    ReflectiveClassBuildItem registerForReflection() {
+        List<Class<?>> classes = List
+                .of(RangeResponse.class, RangeResponse.Builder.class, ResponseHeader.class, ResponseHeader.Builder.class, KeyValue.class);
+        log.info("Registering classes for reflection: " + classes);
+        return ReflectiveClassBuildItem.builder(classes.toArray(new Class<?>[0])).constructors(true).methods(true).fields(true).build();
+    }
+}

--- a/coffee-quarkus-extensions/coffee-module-etcd-extension-parent/coffee-module-etcd-extension/pom.xml
+++ b/coffee-quarkus-extensions/coffee-module-etcd-extension-parent/coffee-module-etcd-extension/pom.xml
@@ -8,25 +8,41 @@
         <version>2.11.0-SNAPSHOT</version>
     </parent>
 
-    <artifactId>coffee-module-etcd-extension-deployment</artifactId>
-    <name>Coff:ee module etcd Extension - Deployment</name>
+    <artifactId>coffee-module-etcd-extension</artifactId>
+    <name>Coff:ee module etcd Extension - Runtime</name>
 
     <dependencies>
 
         <dependency>
             <groupId>io.quarkus</groupId>
-            <artifactId>quarkus-arc-deployment</artifactId>
+            <artifactId>quarkus-arc</artifactId>
         </dependency>
 
         <dependency>
             <groupId>hu.icellmobilsoft.coffee</groupId>
-            <artifactId>coffee-module-etcd-extension</artifactId>
+            <artifactId>coffee-module-etcd</artifactId>
         </dependency>
 
     </dependencies>
 
     <build>
         <plugins>
+            <plugin>
+                <groupId>io.quarkus</groupId>
+                <artifactId>quarkus-extension-maven-plugin</artifactId>
+                <version>${quarkus.version}</version>
+                <executions>
+                    <execution>
+                        <phase>compile</phase>
+                        <goals>
+                            <goal>extension-descriptor</goal>
+                        </goals>
+                        <configuration>
+                            <deployment>${project.groupId}:${project.artifactId}-deployment:${project.version}</deployment>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
             <plugin>
                 <artifactId>maven-compiler-plugin</artifactId>
                 <configuration>

--- a/coffee-quarkus-extensions/coffee-module-etcd-extension-parent/coffee-module-etcd-extension/src/main/resources/META-INF/quarkus-extension.yml
+++ b/coffee-quarkus-extensions/coffee-module-etcd-extension-parent/coffee-module-etcd-extension/src/main/resources/META-INF/quarkus-extension.yml
@@ -1,0 +1,8 @@
+name: "Quarkus Extension for Coffee ETCD"
+artifact: ${project.groupId}:${project.artifactId}:${project.version}
+metadata:
+  keywords:
+    - "coffee"
+    - "coffee-etcd"
+    - "jetcd"
+  status: "experimental"

--- a/coffee-quarkus-extensions/coffee-module-etcd-extension-parent/pom.xml
+++ b/coffee-quarkus-extensions/coffee-module-etcd-extension-parent/pom.xml
@@ -16,6 +16,7 @@
     <description>Quarkus Extensions - Coff:ee module etcd extension</description>
 
     <modules>
+        <module>coffee-module-etcd-extension</module>
         <module>coffee-module-etcd-extension-deployment</module>
     </modules>
 

--- a/coffee-quarkus-extensions/coffee-module-etcd-extension-parent/pom.xml
+++ b/coffee-quarkus-extensions/coffee-module-etcd-extension-parent/pom.xml
@@ -1,0 +1,57 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>hu.icellmobilsoft.coffee</groupId>
+        <artifactId>coffee-quarkus-extensions</artifactId>
+        <version>2.11.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>coffee-module-etcd-extension-parent</artifactId>
+
+    <packaging>pom</packaging>
+
+    <name>Coff:ee module etcd Extension - Parent</name>
+    <description>Quarkus Extensions - Coff:ee module etcd extension</description>
+
+    <modules>
+        <module>coffee-module-etcd-extension-deployment</module>
+    </modules>
+
+    <build>
+        <pluginManagement>
+            <plugins>
+                <plugin>
+                    <groupId>io.quarkus</groupId>
+                    <artifactId>quarkus-maven-plugin</artifactId>
+                    <version>${quarkus.version}</version>
+                </plugin>
+                <plugin>
+                    <artifactId>maven-surefire-plugin</artifactId>
+                    <configuration>
+                        <systemPropertyVariables>
+                            <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
+                        </systemPropertyVariables>
+                    </configuration>
+                </plugin>
+                <plugin>
+                    <artifactId>maven-failsafe-plugin</artifactId>
+                    <configuration>
+                        <systemPropertyVariables>
+                            <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
+                        </systemPropertyVariables>
+                    </configuration>
+                </plugin>
+                <plugin>
+                    <artifactId>maven-compiler-plugin</artifactId>
+                    <configuration>
+                        <compilerArgs>
+                            <arg>-parameters</arg>
+                        </compilerArgs>
+                    </configuration>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+    </build>
+</project>

--- a/coffee-quarkus-extensions/coffee-module-etcd-extension-parent/pom.xml
+++ b/coffee-quarkus-extensions/coffee-module-etcd-extension-parent/pom.xml
@@ -20,39 +20,4 @@
         <module>coffee-module-etcd-extension-deployment</module>
     </modules>
 
-    <build>
-        <pluginManagement>
-            <plugins>
-                <plugin>
-                    <groupId>io.quarkus</groupId>
-                    <artifactId>quarkus-maven-plugin</artifactId>
-                    <version>${quarkus.version}</version>
-                </plugin>
-                <plugin>
-                    <artifactId>maven-surefire-plugin</artifactId>
-                    <configuration>
-                        <systemPropertyVariables>
-                            <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
-                        </systemPropertyVariables>
-                    </configuration>
-                </plugin>
-                <plugin>
-                    <artifactId>maven-failsafe-plugin</artifactId>
-                    <configuration>
-                        <systemPropertyVariables>
-                            <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
-                        </systemPropertyVariables>
-                    </configuration>
-                </plugin>
-                <plugin>
-                    <artifactId>maven-compiler-plugin</artifactId>
-                    <configuration>
-                        <compilerArgs>
-                            <arg>-parameters</arg>
-                        </compilerArgs>
-                    </configuration>
-                </plugin>
-            </plugins>
-        </pluginManagement>
-    </build>
 </project>

--- a/coffee-quarkus-extensions/coffee-module-mp-restclient-extension-parent/pom.xml
+++ b/coffee-quarkus-extensions/coffee-module-mp-restclient-extension-parent/pom.xml
@@ -19,40 +19,4 @@
         <module>coffee-module-mp-restclient-extension</module>
     </modules>
 
-
-    <build>
-        <pluginManagement>
-            <plugins>
-                <plugin>
-                    <groupId>io.quarkus</groupId>
-                    <artifactId>quarkus-maven-plugin</artifactId>
-                    <version>${quarkus.version}</version>
-                </plugin>
-                <plugin>
-                    <artifactId>maven-surefire-plugin</artifactId>
-                    <configuration>
-                        <systemPropertyVariables>
-                            <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
-                        </systemPropertyVariables>
-                    </configuration>
-                </plugin>
-                <plugin>
-                    <artifactId>maven-failsafe-plugin</artifactId>
-                    <configuration>
-                        <systemPropertyVariables>
-                            <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
-                        </systemPropertyVariables>
-                    </configuration>
-                </plugin>
-                <plugin>
-                    <artifactId>maven-compiler-plugin</artifactId>
-                    <configuration>
-                        <compilerArgs>
-                            <arg>-parameters</arg>
-                        </compilerArgs>
-                    </configuration>
-                </plugin>
-            </plugins>
-        </pluginManagement>
-    </build>
 </project>

--- a/coffee-quarkus-extensions/coffee-tool-extension-parent/coffee-tool-extension-deployment/pom.xml
+++ b/coffee-quarkus-extensions/coffee-tool-extension-parent/coffee-tool-extension-deployment/pom.xml
@@ -20,7 +20,7 @@
 
         <dependency>
             <groupId>hu.icellmobilsoft.coffee</groupId>
-            <artifactId>coffee-tool</artifactId>
+            <artifactId>coffee-tool-extension</artifactId>
         </dependency>
 
     </dependencies>

--- a/coffee-quarkus-extensions/coffee-tool-extension-parent/coffee-tool-extension-deployment/pom.xml
+++ b/coffee-quarkus-extensions/coffee-tool-extension-parent/coffee-tool-extension-deployment/pom.xml
@@ -1,0 +1,44 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>hu.icellmobilsoft.coffee</groupId>
+        <artifactId>coffee-tool-extension-parent</artifactId>
+        <version>2.11.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>coffee-tool-extension-deployment</artifactId>
+    <name>Coff:ee tool Extension - Deployment</name>
+
+    <dependencies>
+
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-arc-deployment</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>hu.icellmobilsoft.coffee</groupId>
+            <artifactId>coffee-tool</artifactId>
+        </dependency>
+
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <annotationProcessorPaths>
+                        <path>
+                            <groupId>io.quarkus</groupId>
+                            <artifactId>quarkus-extension-processor</artifactId>
+                            <version>${quarkus.version}</version>
+                        </path>
+                    </annotationProcessorPaths>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+</project>

--- a/coffee-quarkus-extensions/coffee-tool-extension-parent/coffee-tool-extension-deployment/src/main/java/hu/icellmobilsoft/coffee/quarkus/extension/tool/deployment/ToolReflectionProcessor.java
+++ b/coffee-quarkus-extensions/coffee-tool-extension-parent/coffee-tool-extension-deployment/src/main/java/hu/icellmobilsoft/coffee/quarkus/extension/tool/deployment/ToolReflectionProcessor.java
@@ -1,0 +1,59 @@
+/*-
+ * #%L
+ * Coffee
+ * %%
+ * Copyright (C) 2020 - 2025 i-Cell Mobilsoft Zrt.
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package hu.icellmobilsoft.coffee.quarkus.extension.tool.deployment;
+
+import java.util.List;
+
+import org.jboss.logging.Logger;
+
+import hu.icellmobilsoft.coffee.tool.jsonb.FieldOnlyVisibilityStrategy;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+
+/**
+ * Registers necessary coffee-tool related classes for reflection to ensure compatibility with Quarkus native
+ *
+ * @author janos.hamrak
+ * @since 2.11.0
+ */
+public class ToolReflectionProcessor {
+
+    private static final Logger log = Logger.getLogger(ToolReflectionProcessor.class);
+
+    /**
+     * Default constructor
+     */
+    public ToolReflectionProcessor() {
+        // Default constructor for java 21
+    }
+
+    /**
+     * Register coffee-tool classes for reflection.
+     *
+     * @return build item with registered classes
+     */
+    @BuildStep
+    ReflectiveClassBuildItem registerForReflection() {
+        List<Class<?>> classes = List
+                .of(FieldOnlyVisibilityStrategy.class);
+        log.info("Registering classes for reflection: " + classes);
+        return ReflectiveClassBuildItem.builder(classes.toArray(new Class<?>[0])).constructors(true).methods(true).fields(true).build();
+    }
+}

--- a/coffee-quarkus-extensions/coffee-tool-extension-parent/coffee-tool-extension-deployment/src/main/java/hu/icellmobilsoft/coffee/quarkus/extension/tool/deployment/ToolReflectionProcessor.java
+++ b/coffee-quarkus-extensions/coffee-tool-extension-parent/coffee-tool-extension-deployment/src/main/java/hu/icellmobilsoft/coffee/quarkus/extension/tool/deployment/ToolReflectionProcessor.java
@@ -25,6 +25,7 @@ import org.jboss.logging.Logger;
 
 import hu.icellmobilsoft.coffee.tool.jsonb.FieldOnlyVisibilityStrategy;
 import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
 import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
 
 /**
@@ -36,12 +37,18 @@ import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
 public class ToolReflectionProcessor {
 
     private static final Logger log = Logger.getLogger(ToolReflectionProcessor.class);
+    private static final String FEATURE = "coffee-tool-extension";
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
+    }
 
     /**
-     * Default constructor
+     * No-arg constructor
      */
     public ToolReflectionProcessor() {
-        // Default constructor for java 21
+        // Np-arg constructor for java 21
     }
 
     /**

--- a/coffee-quarkus-extensions/coffee-tool-extension-parent/coffee-tool-extension-deployment/src/main/java/hu/icellmobilsoft/coffee/quarkus/extension/tool/deployment/ToolReflectionProcessor.java
+++ b/coffee-quarkus-extensions/coffee-tool-extension-parent/coffee-tool-extension-deployment/src/main/java/hu/icellmobilsoft/coffee/quarkus/extension/tool/deployment/ToolReflectionProcessor.java
@@ -39,16 +39,21 @@ public class ToolReflectionProcessor {
     private static final Logger log = Logger.getLogger(ToolReflectionProcessor.class);
     private static final String FEATURE = "coffee-tool-extension";
 
-    @BuildStep
-    FeatureBuildItem feature() {
-        return new FeatureBuildItem(FEATURE);
-    }
-
     /**
      * No-arg constructor
      */
     public ToolReflectionProcessor() {
         // Np-arg constructor for java 21
+    }
+
+    /**
+     * Registers feature name
+     *
+     * @return Feature build item
+     */
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
     }
 
     /**

--- a/coffee-quarkus-extensions/coffee-tool-extension-parent/coffee-tool-extension/pom.xml
+++ b/coffee-quarkus-extensions/coffee-tool-extension-parent/coffee-tool-extension/pom.xml
@@ -4,29 +4,45 @@
     <modelVersion>4.0.0</modelVersion>
     <parent>
         <groupId>hu.icellmobilsoft.coffee</groupId>
-        <artifactId>coffee-module-etcd-extension-parent</artifactId>
+        <artifactId>coffee-tool-extension-parent</artifactId>
         <version>2.11.0-SNAPSHOT</version>
     </parent>
 
-    <artifactId>coffee-module-etcd-extension-deployment</artifactId>
-    <name>Coff:ee module etcd Extension - Deployment</name>
+    <artifactId>coffee-tool-extension</artifactId>
+    <name>Coff:ee tool Extension - Runtime</name>
 
     <dependencies>
 
         <dependency>
             <groupId>io.quarkus</groupId>
-            <artifactId>quarkus-arc-deployment</artifactId>
+            <artifactId>quarkus-arc</artifactId>
         </dependency>
 
         <dependency>
             <groupId>hu.icellmobilsoft.coffee</groupId>
-            <artifactId>coffee-module-etcd-extension</artifactId>
+            <artifactId>coffee-tool</artifactId>
         </dependency>
 
     </dependencies>
 
     <build>
         <plugins>
+            <plugin>
+                <groupId>io.quarkus</groupId>
+                <artifactId>quarkus-extension-maven-plugin</artifactId>
+                <version>${quarkus.version}</version>
+                <executions>
+                    <execution>
+                        <phase>compile</phase>
+                        <goals>
+                            <goal>extension-descriptor</goal>
+                        </goals>
+                        <configuration>
+                            <deployment>${project.groupId}:${project.artifactId}-deployment:${project.version}</deployment>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
             <plugin>
                 <artifactId>maven-compiler-plugin</artifactId>
                 <configuration>

--- a/coffee-quarkus-extensions/coffee-tool-extension-parent/coffee-tool-extension/src/main/resources/META-INF/quarkus-extension.yml
+++ b/coffee-quarkus-extensions/coffee-tool-extension-parent/coffee-tool-extension/src/main/resources/META-INF/quarkus-extension.yml
@@ -1,0 +1,8 @@
+name: "Quarkus Extension for Coffee tool"
+artifact: ${project.groupId}:${project.artifactId}:${project.version}
+metadata:
+  keywords:
+    - "coffee"
+    - "coffee-tool"
+    - "jsonb"
+  status: "experimental"

--- a/coffee-quarkus-extensions/coffee-tool-extension-parent/pom.xml
+++ b/coffee-quarkus-extensions/coffee-tool-extension-parent/pom.xml
@@ -15,6 +15,7 @@
     <description>Quarkus Extensions - Coff:ee tool extension</description>
 
     <modules>
+        <module>coffee-tool-extension</module>
         <module>coffee-tool-extension-deployment</module>
     </modules>
 

--- a/coffee-quarkus-extensions/coffee-tool-extension-parent/pom.xml
+++ b/coffee-quarkus-extensions/coffee-tool-extension-parent/pom.xml
@@ -19,39 +19,4 @@
         <module>coffee-tool-extension-deployment</module>
     </modules>
 
-    <build>
-        <pluginManagement>
-            <plugins>
-                <plugin>
-                    <groupId>io.quarkus</groupId>
-                    <artifactId>quarkus-maven-plugin</artifactId>
-                    <version>${quarkus.version}</version>
-                </plugin>
-                <plugin>
-                    <artifactId>maven-surefire-plugin</artifactId>
-                    <configuration>
-                        <systemPropertyVariables>
-                            <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
-                        </systemPropertyVariables>
-                    </configuration>
-                </plugin>
-                <plugin>
-                    <artifactId>maven-failsafe-plugin</artifactId>
-                    <configuration>
-                        <systemPropertyVariables>
-                            <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
-                        </systemPropertyVariables>
-                    </configuration>
-                </plugin>
-                <plugin>
-                    <artifactId>maven-compiler-plugin</artifactId>
-                    <configuration>
-                        <compilerArgs>
-                            <arg>-parameters</arg>
-                        </compilerArgs>
-                    </configuration>
-                </plugin>
-            </plugins>
-        </pluginManagement>
-    </build>
 </project>

--- a/coffee-quarkus-extensions/coffee-tool-extension-parent/pom.xml
+++ b/coffee-quarkus-extensions/coffee-tool-extension-parent/pom.xml
@@ -1,0 +1,56 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>hu.icellmobilsoft.coffee</groupId>
+        <artifactId>coffee-quarkus-extensions</artifactId>
+        <version>2.11.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>coffee-tool-extension-parent</artifactId>
+
+    <packaging>pom</packaging>
+
+    <name>Coff:ee tool Extension - Parent</name>
+    <description>Quarkus Extensions - Coff:ee tool extension</description>
+
+    <modules>
+        <module>coffee-tool-extension-deployment</module>
+    </modules>
+
+    <build>
+        <pluginManagement>
+            <plugins>
+                <plugin>
+                    <groupId>io.quarkus</groupId>
+                    <artifactId>quarkus-maven-plugin</artifactId>
+                    <version>${quarkus.version}</version>
+                </plugin>
+                <plugin>
+                    <artifactId>maven-surefire-plugin</artifactId>
+                    <configuration>
+                        <systemPropertyVariables>
+                            <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
+                        </systemPropertyVariables>
+                    </configuration>
+                </plugin>
+                <plugin>
+                    <artifactId>maven-failsafe-plugin</artifactId>
+                    <configuration>
+                        <systemPropertyVariables>
+                            <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
+                        </systemPropertyVariables>
+                    </configuration>
+                </plugin>
+                <plugin>
+                    <artifactId>maven-compiler-plugin</artifactId>
+                    <configuration>
+                        <compilerArgs>
+                            <arg>-parameters</arg>
+                        </compilerArgs>
+                    </configuration>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+    </build>
+</project>

--- a/coffee-quarkus-extensions/pom.xml
+++ b/coffee-quarkus-extensions/pom.xml
@@ -17,6 +17,8 @@
     <modules>
         <module>coffee-module-mp-restclient-extension-parent</module>
         <module>coffee-deltaspike-data-extension-parent</module>
+        <module>coffee-module-etcd-extension-parent</module>
+        <module>coffee-tool-extension-parent</module>
     </modules>
 
     <dependencyManagement>

--- a/coffee-quarkus-extensions/pom.xml
+++ b/coffee-quarkus-extensions/pom.xml
@@ -40,4 +40,40 @@
         </dependencies>
     </dependencyManagement>
 
+    <build>
+        <pluginManagement>
+            <plugins>
+                <plugin>
+                    <groupId>io.quarkus</groupId>
+                    <artifactId>quarkus-maven-plugin</artifactId>
+                    <version>${quarkus.version}</version>
+                </plugin>
+                <plugin>
+                    <artifactId>maven-surefire-plugin</artifactId>
+                    <configuration>
+                        <systemPropertyVariables>
+                            <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
+                        </systemPropertyVariables>
+                    </configuration>
+                </plugin>
+                <plugin>
+                    <artifactId>maven-failsafe-plugin</artifactId>
+                    <configuration>
+                        <systemPropertyVariables>
+                            <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
+                        </systemPropertyVariables>
+                    </configuration>
+                </plugin>
+                <plugin>
+                    <artifactId>maven-compiler-plugin</artifactId>
+                    <configuration>
+                        <compilerArgs>
+                            <arg>-parameters</arg>
+                        </compilerArgs>
+                    </configuration>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+    </build>
+
 </project>

--- a/docs/en/common/coffee-quarkus-extensions.adoc
+++ b/docs/en/common/coffee-quarkus-extensions.adoc
@@ -13,3 +13,5 @@ Quarkus extension building basics help: https://quarkus.io/guides/building-my-fi
 List of extensions built specifically for Quarkus: https://quarkus.pro/extensions[Link]
 
 include::quarkus-extension/coffee-module-mp-restclient-extension.adoc[leveloffset=+1]
+include::quarkus-extension/coffee-module-etcd-extension.adoc[leveloffset=+1]
+include::quarkus-extension/coffee-tool-extension.adoc[leveloffset=+1]

--- a/docs/en/common/quarkus-extension/coffee-module-etcd-extension.adoc
+++ b/docs/en/common/quarkus-extension/coffee-module-etcd-extension.adoc
@@ -15,7 +15,7 @@ To use this module, you should add the extension dependency instead of `coffee-m
 ----
     <dependency>
         <groupId>hu.icellmobilsoft.coffee</groupId>
-        <artifactId>coffee-module-etcd-extension-deployment</artifactId>
+        <artifactId>coffee-module-etcd-extension</artifactId>
         <version>${version.hu.icellmobilsoft.coffee}</version>
     </dependency>
 ----

--- a/docs/en/common/quarkus-extension/coffee-module-etcd-extension.adoc
+++ b/docs/en/common/quarkus-extension/coffee-module-etcd-extension.adoc
@@ -1,0 +1,21 @@
+[#common_coffee-quarkus-extensions-module-etcd]
+= coffee-module-etcd-extension
+
+The purpose of this module is to extend the `coffee-module-etcd` module with Quarkus native runtime support.
+
+The Quarkus native compiler (GraalVM) removes classes at build time that are not directly referenced.
+This extension explicitly registers the classes referenced via reflection in the `coffee-module-etcd` module, to prevent them from being removed by the compiler.
+
+NOTE: currently, version 3.2.x of Quarkus is used in the extensions.
+
+To use this module, you should add the extension dependency instead of `coffee-module-etcd`, as it already includes the `coffee-module-etcd` module.
+
+.pom.xml
+[source,xml]
+----
+    <dependency>
+        <groupId>hu.icellmobilsoft.coffee</groupId>
+        <artifactId>coffee-module-etcd-extension-deployment</artifactId>
+        <version>${version.hu.icellmobilsoft.coffee}</version>
+    </dependency>
+----

--- a/docs/en/common/quarkus-extension/coffee-tool-extension.adoc
+++ b/docs/en/common/quarkus-extension/coffee-tool-extension.adoc
@@ -15,7 +15,7 @@ To use this module, you should add the extension dependency instead of `coffee-t
 ----
     <dependency>
         <groupId>hu.icellmobilsoft.coffee</groupId>
-        <artifactId>coffee-tool-extension-deployment</artifactId>
+        <artifactId>coffee-tool-extension</artifactId>
         <version>${version.hu.icellmobilsoft.coffee}</version>
     </dependency>
 ----

--- a/docs/en/common/quarkus-extension/coffee-tool-extension.adoc
+++ b/docs/en/common/quarkus-extension/coffee-tool-extension.adoc
@@ -1,0 +1,21 @@
+[#common_coffee-quarkus-extensions-tool]
+= coffee-tool-extension
+
+The purpose of this module is to extend the `coffee-tool` module with Quarkus native runtime support.
+
+The Quarkus native compiler (GraalVM) removes classes at build time that are not directly referenced.
+This extension explicitly registers the classes referenced via reflection in the `coffee-tool` module, to prevent them from being removed by the compiler.
+
+NOTE: currently, version 3.2.x of Quarkus is used in the extensions.
+
+To use this module, you should add the extension dependency instead of `coffee-tool`, as it already includes the `coffee-tool` module.
+
+.pom.xml
+[source,xml]
+----
+    <dependency>
+        <groupId>hu.icellmobilsoft.coffee</groupId>
+        <artifactId>coffee-tool-extension-deployment</artifactId>
+        <version>${version.hu.icellmobilsoft.coffee}</version>
+    </dependency>
+----

--- a/docs/en/migration/migration2100to2110.adoc
+++ b/docs/en/migration/migration2100to2110.adoc
@@ -18,3 +18,13 @@ Hibernate `6.2.13.Final` -> `6.6.1.Final` upgrade (Wildfly 34 support).
 ==== Migration
 * https://docs.jboss.org/hibernate/orm/6.6/migration-guide/migration-guide.html[Hibernate 6.6 migration guide]
 * Otherwise, no changes were made at coff:ee that would prevent the use of the previous hibernate version with a dependency exclude
+
+=== coffee-module-etcd-extension-deployment
+
+New Quarkus extension to support Quarkus native compatibility for the coffee-module-etcd module.
+Description: <<common_coffee-quarkus-extensions-module-etcd, coffee-module-etcd-extension>>
+
+=== coffee-tool-extension-deployment
+
+New Quarkus extension to support Quarkus native compatibility for the coffee-tool module.
+Description: <<common_coffee-quarkus-extensions-tool, coffee-tool-extension>>

--- a/docs/en/migration/migration2100to2110.adoc
+++ b/docs/en/migration/migration2100to2110.adoc
@@ -19,12 +19,12 @@ Hibernate `6.2.13.Final` -> `6.6.1.Final` upgrade (Wildfly 34 support).
 * https://docs.jboss.org/hibernate/orm/6.6/migration-guide/migration-guide.html[Hibernate 6.6 migration guide]
 * Otherwise, no changes were made at coff:ee that would prevent the use of the previous hibernate version with a dependency exclude
 
-=== coffee-module-etcd-extension-deployment
+=== coffee-module-etcd-extension
 
 New Quarkus extension to support Quarkus native compatibility for the coffee-module-etcd module.
 Description: <<common_coffee-quarkus-extensions-module-etcd, coffee-module-etcd-extension>>
 
-=== coffee-tool-extension-deployment
+=== coffee-tool-extension
 
 New Quarkus extension to support Quarkus native compatibility for the coffee-tool module.
 Description: <<common_coffee-quarkus-extensions-tool, coffee-tool-extension>>

--- a/docs/hu/common/coffee-quarkus-extensions.adoc
+++ b/docs/hu/common/coffee-quarkus-extensions.adoc
@@ -13,3 +13,5 @@ Quarkus extension készítés alapok segédlet: https://quarkus.io/guides/buildi
 Quarkus-hoz konkrétan elkészítésett Extensionök listája: https://quarkus.pro/extensions[Link]
 
 include::quarkus-extension/coffee-module-mp-restclient-extension.adoc[leveloffset=+1]
+include::quarkus-extension/coffee-module-etcd-extension.adoc[leveloffset=+1]
+include::quarkus-extension/coffee-tool-extension.adoc[leveloffset=+1]

--- a/docs/hu/common/quarkus-extension/coffee-module-etcd-extension.adoc
+++ b/docs/hu/common/quarkus-extension/coffee-module-etcd-extension.adoc
@@ -15,7 +15,7 @@ A modul használatához a `coffee-module-etcd` helyett az extension függősége
 ----
     <dependency>
         <groupId>hu.icellmobilsoft.coffee</groupId>
-        <artifactId>coffee-module-etcd-extension-deployment</artifactId>
+        <artifactId>coffee-module-etcd-extension</artifactId>
         <version>${version.hu.icellmobilsoft.coffee}</version>
     </dependency>
 ----

--- a/docs/hu/common/quarkus-extension/coffee-module-etcd-extension.adoc
+++ b/docs/hu/common/quarkus-extension/coffee-module-etcd-extension.adoc
@@ -1,0 +1,21 @@
+[#common_coffee-quarkus-extensions-module-etcd]
+= coffee-module-etcd-extension
+
+A modul céja a `coffee-module-etcd` modul kiterjesztése Quarkus natív futtatás támogatással.
+
+A Quarkus natív fordító (GraalVM) fordítási időben eltávolítja azokat az osztályokat, amelyekre nincs közvetlen hivatkozás.
+Az extension a `coffee-module-etcd` modulban található reflection-nel hivatkozott osztályokat explicit regisztrálja, hogy ne kerüljenek törlésre a fordító által.
+
+NOTE: jelenleg a Quarkus 3.2.x verziója van használva az extension-ökben.
+
+A modul használatához a `coffee-module-etcd` helyett az extension függőséget kell hozzáadni, ez már tartalmazza magát a `coffee-module-etcd` modult is.
+
+.pom.xml
+[source,xml]
+----
+    <dependency>
+        <groupId>hu.icellmobilsoft.coffee</groupId>
+        <artifactId>coffee-module-etcd-extension-deployment</artifactId>
+        <version>${version.hu.icellmobilsoft.coffee}</version>
+    </dependency>
+----

--- a/docs/hu/common/quarkus-extension/coffee-tool-extension.adoc
+++ b/docs/hu/common/quarkus-extension/coffee-tool-extension.adoc
@@ -15,7 +15,7 @@ A modul használatához a `coffee-tool` helyett az extension függőséget kell 
 ----
     <dependency>
         <groupId>hu.icellmobilsoft.coffee</groupId>
-        <artifactId>coffee-tool-extension-deployment</artifactId>
+        <artifactId>coffee-tool-extension</artifactId>
         <version>${version.hu.icellmobilsoft.coffee}</version>
     </dependency>
 ----

--- a/docs/hu/common/quarkus-extension/coffee-tool-extension.adoc
+++ b/docs/hu/common/quarkus-extension/coffee-tool-extension.adoc
@@ -1,0 +1,21 @@
+[#common_coffee-quarkus-extensions-tool]
+= coffee-tool-extension
+
+A modul céja a `coffee-tool` modul kiterjesztése Quarkus natív futtatás támogatással.
+
+A Quarkus natív fordító (GraalVM) fordítási időben eltávolítja azokat az osztályokat, amelyekre nincs közvetlen hivatkozás.
+Az extension a `coffee-tool` modulban található reflection-nel hivatkozott osztályokat explicit regisztrálja, hogy ne kerüljenek törlésre a fordító által.
+
+NOTE: jelenleg a Quarkus 3.2.x verziója van használva az extension-ökben.
+
+A modul használatához a `coffee-tool` helyett az extension függőséget kell hozzáadni, ez már tartalmazza magát a `coffee-tool` modult is.
+
+.pom.xml
+[source,xml]
+----
+    <dependency>
+        <groupId>hu.icellmobilsoft.coffee</groupId>
+        <artifactId>coffee-tool-extension-deployment</artifactId>
+        <version>${version.hu.icellmobilsoft.coffee}</version>
+    </dependency>
+----

--- a/docs/hu/migration/migration2100to2110.adoc
+++ b/docs/hu/migration/migration2100to2110.adoc
@@ -18,12 +18,12 @@ Hibernate `6.2.13.Final` -> `6.6.1.Final` emelés (Wildfly 34 támogatás).
 * https://docs.jboss.org/hibernate/orm/6.6/migration-guide/migration-guide.html[Hibernate 6.6 migration guide]
 * Egyébként nem történt olyan válatoztatás coff:ee szinten, ami gátolná dependency exclude-al a korábbi hibernate verzió használatát
 
-=== coffee-module-etcd-extension-deployment
+=== coffee-module-etcd-extension
 
 Új Quarkus extension a coffee-module-etcd modul Quarkus natív kompatibilitás támogatásához.
 Leírás: <<common_coffee-quarkus-extensions-module-etcd, coffee-module-etcd-extension>>
 
-=== coffee-tool-extension-deployment
+=== coffee-tool-extension
 
 Új Quarkus extension a coffee-tool modul Quarkus natív kompatibilitás támogatásához.
 Leírás: <<common_coffee-quarkus-extensions-tool, coffee-module-etcd-extension>>

--- a/docs/hu/migration/migration2100to2110.adoc
+++ b/docs/hu/migration/migration2100to2110.adoc
@@ -17,3 +17,13 @@ Hibernate `6.2.13.Final` -> `6.6.1.Final` emelés (Wildfly 34 támogatás).
 ==== Migration
 * https://docs.jboss.org/hibernate/orm/6.6/migration-guide/migration-guide.html[Hibernate 6.6 migration guide]
 * Egyébként nem történt olyan válatoztatás coff:ee szinten, ami gátolná dependency exclude-al a korábbi hibernate verzió használatát
+
+=== coffee-module-etcd-extension-deployment
+
+Új Quarkus extension a coffee-module-etcd modul Quarkus natív kompatibilitás támogatásához.
+Leírás: <<common_coffee-quarkus-extensions-module-etcd, coffee-module-etcd-extension>>
+
+=== coffee-tool-extension-deployment
+
+Új Quarkus extension a coffee-tool modul Quarkus natív kompatibilitás támogatásához.
+Leírás: <<common_coffee-quarkus-extensions-tool, coffee-module-etcd-extension>>

